OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/ResetPasswordAlertDialogFragment.java

```diff
@@ -0,0 +1,200 @@
+            @Override
+            public void onClick(View v) {
+                submit(getEnteredEmail());
+            }
+        });
+    }
+
+
+    @Override
+    public void onDismiss(DialogInterface dialog) {
+        super.onDismiss(dialog);
+        if (listener!=null) {
+            listener.omDismissed();
+        }
+    }
+
+    public void setListener(@Nullable Listener listener) {
+        this.listener = listener;
+    }
+
+    @Nullable
+    protected String getEnteredEmail() {
+        String email = null;
+
+        if (et != null) {
+            email = et.getText().toString().trim();
+        }
+
+        return email;
+    }
+
+    /**
+     * Shows or hides error
+     * @param error The error to show. If null, error will be hidden
+     */
+    public void showError(@Nullable String error) {
+        if (til != null) {
+            if (error != null) {
+                til.setError(error);
+                til.setErrorEnabled(true);
+                til.sendAccessibilityEvent(AccessibilityEvent.TYPE_VIEW_FOCUSED);
+            } else {
+                til.setErrorEnabled(false);
+            }
+        }
+    }
+
+    private void submit(@Nullable String email) {
+        if (!NetworkUtil.isConnected(getContext())) {
+            showError(getString(R.string.network_not_connected_short));
+        } else if (!InputValidationUtil.isValidEmail(email)) {
+            showError(getString(R.string.error_invalid_email));
+        } else {
+            loginService.resetPassword(email).enqueue(new ErrorHandlingCallback<ResetPasswordResponse>(
+                    getContext(), CallTrigger.USER_ACTION) {
+                @Override
+                protected void onResponse(@NonNull final ResetPasswordResponse result) {
+                    if (listener!=null) {
+                        if (result.isSuccess()) {
+                            listener.onResult(true, null);
+                        }
+                        else {
+                            listener.onResult(false, result.getPrimaryReason());
+                        }
+                    }
+                }
+
+                @Override
+                protected void onFailure(@NonNull Throwable error) {
+                    if (listener != null) {
+                        listener.onResult(false, ErrorUtils.getErrorMessage(error, getContext()));
+                    }
+                }
+            });
+        }
+    }
+
+    @Nullable
+    @Override
+    protected ButtonAttributes getNegativeButtonAttributes() {
+        return new ButtonAttributes() {
+            @NonNull
+            @Override
+            public String getMessage() {
+                return getContext().getResources().getString(R.string.label_cancel);
+            }
+
+            @Nullable
+            @Override
+            public DialogInterface.OnClickListener getOnClickListener() {
+                return null;
```

Comment:
No, for the positve button, we are setting to null temporarily (I'll update that comment to be more clear). But for  the negative button we are leaving it as null